src/NCronJob/Scheduler/QueueWorker.cs

```diff
@@ -57,10 +60,14 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         stopToken.Register(LogCancellationRequestedInJob);
         var runningTasks = new List<Task>();
 
+        await startupJobManager.ProcessStartupJobs(CreateExecutionTask, stopToken);
+
         ScheduleInitialJobs();
 
         try
         {
+            await startupJobManager.WaitForStartupJobsCompletion();
```

Comment:
That is a very good point - migrations and so on should be executed before anything else (even other jobs are started).

The "issue" is that they bypass our other boundaries like retry-policy/concurrency control.

Reply:
Actually retry will be honored since that's handled by the `IRetryHandler` which happens in the `JobExecutor`. However concurrency management will not. But realistically what are the odds that you would need concurrency management within startup jobs? They won't have to compete for resources with regular jobs (given that they need to finish before any normal jobs start up).

Reply:
And I guess that since they only run once then you can't (or shouldn't) overlap them like this:
```csharp
builder.Services.AddNCronJob(b =>
    b.AddJob<RunAtStartJob>()
        .RunAtStartup()
        .RunAtStartup());

builder.Services.AddNCronJob(b => b.AddJob<RunAtStartJob>(p =>
    {
        p.RunAtStartup();
        p.RunAtStartup();
    }
));

builder.Services.AddNCronJob(b => b
    .AddJob<RunAtStartJob>(p =>
        p.RunAtStartup())
    .AddJob<RunAtStartJob>(p =>
        p.RunAtStartup())
);
```

Reply:
I'm dropping support for chaining on to `JobOptionBuilder`, seems too messy and confusing. This will prevent someone from doing this:

```csharp
builder.Services.AddNCronJob(b => b.AddJob<RunAtStartJob>(p =>
    {
        p.RunAtStartup();
        p.RunAtStartup();
    }
));
...
builder.Services.AddNCronJob(b => b
    .AddJob<RunAtStartJob>(p =>
        p.RunAtStartup())
    .AddJob<RunAtStartJob>(p =>
        p.RunAtStartup())
);
```

The API should guide users to set it up in only one way, from `NCronJobOptionBuilder<TJob>` like this:
```csharp
builder.Services.AddNCronJob(b => b.AddJob<RunAtStartJob>().RunAtStartup());
...
builder.Services.AddNCronJob(options =>
{
    options.AddJob<RunAtStartJob>()
        .RunAtStartup()
        .AddNotificationHandler<RunAtStartJobHandler>();
});
```

Reply:
That is a very good idea!

Reply:
This commit enhances the API to guide the dev
https://github.com/NCronJob-Dev/NCronJob/pull/70/commits/8b4525df3d71edad0975590d33dcf0710b885153